applications/spark/README.md

```diff
@@ -2,51 +2,170 @@
+    ```
+
+    **Note**: This command creates several directories that are used by the cluster. If you will
+    run multiple clusters simultaneously, be sure to use different base directories. The base
+    directory is the current directory by default, and can be changed with `--directory`.
+
+3. Set this environment variable so that your jobs use the Spark configuration settings that you
+just created.
+    ```
+    $ export SPARK_CONF_DIR=$(pwd)/conf
+    ```
+
+4. Run jobs as described [below](#run-jobs).
+
+## Advanced Configuration Instructions
+This section describes how to run the configuration scripts individually so that you can customize
+any Spark parameter.
+
+1. Acquire compute nodes as discussed above. The rest of the instructions assume that you are
+logged into the first node in the allocation (either through `ssh` or automatically through
+`salloc`).
+
+2. Create a local copy of the configuration files with the command below. Specify an alternate
+destination directory with `-d <directory>`.
+
+    ```
+    $ create_config.sh -c <path-to-spark-container>
+    ```
+
+3. Edit the `config` file if necessary. Note that the rest of this section relies on the setting
 `container_instance_name = spark`.
-6. Consider what type of compute nodes to acquire. If you will be performing large shuffles
-then you must get nodes with fast local storage. `bigmem` and `gpu` nodes have local SSDs that
-can read/write at 2 GB/s. The standard nodes have spinning disks that can only read/write at
-~130 MB/s. Your jobs will fail if you use those nodes. You can consider specifying a RAM disk
-as Spark local storage (`/dev/shm`), but you must be sure you have enough space.
-7. Decide how and when you want to configure your Spark application parameters.
-
-   - Manually specify global settings in `conf`. Note that worker settings in `conf/spark-env.sh`
-   must be set before starting the cluster.
-   - Auto-configure global settings with `configure_spark.sh`. You can run this script after
-   acquiring compute nodes and it will apply settings based on the hardware resources (memory/CPU)
-   of those nodes. Run `configure_spark.sh --help` to see available options.
-   - At runtime when you run `spark-submit` or `pyspark`. Refer to the CLI help.
+
+4. Optional but recommended: Auto-configure the Spark parameters with `configure_spark.sh`. This
+   will adjust the parameters based on the actual memory and CPU resources in your compute nodes.
+   You can run `configure_spark.sh --help` to see all options.
+
+    ```
+    $ configure_spark.sh 
+    ```
+
+5. Optional: Customize the Spark configuration parameters.
+
+This can be done by:
+
+  - Editing the files in the `conf` directory.
+  - Setting the parameters as CLI options when you run `spark-submit` or `pyspark`. Refer to the
+  CLI help.
+
+**Note**: The only way to customize Spark worker parameters is by editing `conf/spark-env.sh`.
```

Comment:
Once you migrate this to the gh-pages branch, we will need to change the notes to use [admonitions](https://docusaurus.io/docs/markdown-features/admonitions).